packages/smooth_app/lib/pages/product/product_image_swipeable_view.dart

```diff
@@ -0,0 +1,79 @@
+import 'package:flutter/material.dart';
+import 'package:smooth_app/data_models/product_image_data.dart';
+import 'package:smooth_app/generic_lib/design_constants.dart';
+import 'package:smooth_app/generic_lib/widgets/smooth_back_button.dart';
+import 'package:smooth_app/pages/product/product_image_viewer.dart';
+
+///Widget to display swipeable product images,
+///Opens product image with [initialProductImageDataIndex] from list of images Typecasted from [selectedImages]
+///
+///Field [selectedImages],[initialProductImageDataIndex],[barcode] cannot be null
+class ProductImageSwipeableView extends StatefulWidget {
+  const ProductImageSwipeableView({
+    Key? key,
+    required this.selectedImages,
+    required this.initialProductImageDataIndex,
+    required this.barcode,
+  }) : super(key: key);
+
+  final Map<ProductImageData, ImageProvider?> selectedImages;
+  final int initialProductImageDataIndex;
+  final String barcode;
+
+  @override
+  State<ProductImageSwipeableView> createState() =>
+      _ProductImageSwipeableViewState();
+}
+
+class _ProductImageSwipeableViewState extends State<ProductImageSwipeableView> {
+  final ValueNotifier<int> currentImageDataIndex = ValueNotifier<int>(0);
```

Comment:
The thing was, I tried using setState , But its lagging due to entire rendering of the page, And that lag was quite noticable.

Reply:
I understand how it could be faster with `ValueNotifier` instead of `setState`, but to be honest I'm surprised that it's so noticeable because there's basically only one extra widget to refresh with `setState`: `ProductImageViewer` (that is probably not well optimized).
Please add an explicit comment about your choice, because later developers may ask the same question: "Why not a `setState`?".
In addition to that,  now that you don't use `setState`, you can convert the widget back to stateless, can't you?